KameletBinding error handler nil map

Using version `1.5` and `latest` the operator may panic when trying to use a KameletBinding error handler. This is the logged exception:
```
{"level":"error","ts":1627032308.0019584,"logger":"cmd","msg":"Observed a panic: \"assignment to entry in nil map\" (assignment to entry in nil map)\ngoroutine 527 [running]:\nk8s.io/apimachinery/pkg/util/runtime.logPanic(0x2132c00, 0x27af5e0)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:74 +0x95\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:48 +0x89\npanic(0x2132c00, 0x27af5e0)\n\truntime/panic.go:969 +0x175\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.setErrorHandlerConfiguration(0xc000410630, 0x281b7c0, 0xc0011eb4f0, 0xc00032dfc0, 0xc0012c5c20)\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:108 +0x18e\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.maybeErrorHandler(0xc0009568c0, 0x2834800, 0xc00043c4e0, 0x2873b80, 0xc00032dfc0, 0xc0012c5c20, 0xd, 0x2479506, 0xa, 0x4, ...)\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:50 +0x11e\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.createIntegrationFor(0x2834800, 0xc00043c4e0, 0x2873b80, 0xc00032dfc0, 0xc0007fb500, 0x0, 0x0, 0x0)\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/common.go:98 +0x625\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.(*initializeAction).Handle(0xc0009567c0, 0x2834800, 0xc00043c4e0, 0xc0007fb500, 0xc0005e3ae8, 0x1, 0x1)\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/initialize.go:53 +0x59\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.(*ReconcileKameletBinding).Reconcile(0xc000b4b8f0, 0x2834800, 0xc00043c4e0, 0xc0012c5c20, 0xd, 0xc0012c5c00, 0xc, 0xc000289ec0, 0xc000289ec0, 0x27e38c0, ...)\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/kamelet_binding_controller.go:177 +0x6d0\ngithub.com/apache/camel-k/pkg/util/monitoring.(*instrumentedReconciler).Reconcile(0xc000af2040, 0x2834800, 0xc00043c4e0, 0xc0012c5c20, 0xd, 0xc0012c5c00, 0xc, 0xc00043c4e0, 0x40a45f, 0xc00003a000, ...)\n\tgithub.com/apache/camel-k/pkg/util/monitoring/controller.go:71 +0xd9\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000c31900, 0x2834740, 0xc000cfb840, 0x21e4d60, 0xc000956540)\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298 +0x317\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000c31900, 0x2834740, 0xc000cfb840, 0x462b367934584500)\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253 +0x205\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2(0x2834740, 0xc000cfb840)\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216 +0x4a\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1()\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185 +0x37\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00029af50)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x5f\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0005e3f50, 0x27e1ec0, 0xc000dfe270, 0xc000cfb801, 0xc00049c060)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0xad\nk8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00029af50, 0x3b9aca00, 0x0, 0x314551776e4d7101, 0xc00049c060)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x98\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext(0x2834740, 0xc000cfb840, 0xc000ccc160, 0x3b9aca00, 0x0, 0x6a716548424c0a01)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185 +0xa6\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext(0x2834740, 0xc000cfb840, 0xc000ccc160, 0x3b9aca00)\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99 +0x57\ncreated by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:213 +0x45e\n","stacktrace":"k8s.io/klog/v2.(*loggingT).output\n\tk8s.io/klog/v2@v2.8.0/klog.go:914\nk8s.io/klog/v2.(*loggingT).printf\n\tk8s.io/klog/v2@v2.8.0/klog.go:751\nk8s.io/klog/v2.Errorf\n\tk8s.io/klog/v2@v2.8.0/klog.go:1463\nk8s.io/apimachinery/pkg/util/runtime.logPanic\n\tk8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:78\nk8s.io/apimachinery/pkg/util/runtime.HandleCrash\n\tk8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:48\nruntime.gopanic\n\truntime/panic.go:969\nruntime.mapassign_faststr\n\truntime/map_faststr.go:204\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.setErrorHandlerConfiguration\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:108\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.maybeErrorHandler\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:50\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.createIntegrationFor\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/common.go:98\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.(*initializeAction).Handle\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/initialize.go:53\ngithub.com/apache/camel-k/pkg/controller/kameletbinding.(*ReconcileKameletBinding).Reconcile\n\tgithub.com/apache/camel-k/pkg/controller/kameletbinding/kamelet_binding_controller.go:177\ngithub.com/apache/camel-k/pkg/util/monitoring.(*instrumentedReconciler).Reconcile\n\tgithub.com/apache/camel-k/pkg/util/monitoring/controller.go:71\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2\n\tsigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\tk8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:99"}
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 527 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x2132c00, 0x27af5e0)
	runtime/panic.go:969 +0x175
github.com/apache/camel-k/pkg/controller/kameletbinding.setErrorHandlerConfiguration(0xc000410630, 0x281b7c0, 0xc0011eb4f0, 0xc00032dfc0, 0xc0012c5c20)
	github.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:108 +0x18e
github.com/apache/camel-k/pkg/controller/kameletbinding.maybeErrorHandler(0xc0009568c0, 0x2834800, 0xc00043c4e0, 0x2873b80, 0xc00032dfc0, 0xc0012c5c20, 0xd, 0x2479506, 0xa, 0x4, ...)
	github.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:50 +0x11e
github.com/apache/camel-k/pkg/controller/kameletbinding.createIntegrationFor(0x2834800, 0xc00043c4e0, 0x2873b80, 0xc00032dfc0, 0xc0007fb500, 0x0, 0x0, 0x0)
```

--- a/pkg/controller/kameletbinding/error_handler.go
+++ b/pkg/controller/kameletbinding/error_handler.go
@@ -100,6 +100,10 @@ func setErrorHandlerConfiguration(errorHandlerBinding *bindings.Binding, errorHa
 	if err != nil {
 		return err
 	}
+	// initialize map if not yet initialized
+	if errorHandlerBinding.ApplicationProperties == nil {
+		errorHandlerBinding.ApplicationProperties = make(map[string]string)
+	}
 	for key, value := range properties {
 		errorHandlerBinding.ApplicationProperties[key] = fmt.Sprintf("%v", value)
 	}

